smart-contracts/test/testCosmosBridge.js

```diff
@@ -306,5 +284,124 @@ describe("Test Cosmos Bridge", function () {
+        sender: state.sender,
+        senderSequence: state.senderSequence,
+        recipientAddress: state.recipient,
+        tokenAddress: state.token.address,
+        amount: state.amount,
+        isDoublePeg: false,
+        nonce: state.nonce,
+        networkDescriptor: state.networkDescriptor,
+        tokenName: state.name,
+        tokenSymbol: state.symbol,
+        tokenDecimals: state.decimals,
+        validators: [userOne, userTwo, userFour],
+      });
+
+      await state.cosmosBridge
+        .connect(userOne)
+        .submitProphecyClaimAggregatedSigs(digest, claimData, signatures);
+
+      const newlyCreatedTokenAddress = await state.cosmosBridge.sourceAddressToDestinationAddress(state.token.address);
+      expect(newlyCreatedTokenAddress).to.be.equal(state.constants.zeroAddress);
+    });
+
+    it("should NOT deploy a new token upon the successful processing of a double-pegged burn prophecy claim for an already managed token", async function () {
+      state.nonce = 1;
+
+      const { digest, claimData, signatures } = await getValidClaim({
+        state,
+        sender: state.sender,
+        senderSequence: state.senderSequence,
+        recipientAddress: state.recipient,
+        tokenAddress: state.token.address,
+        amount: state.amount,
+        isDoublePeg: true,
+        nonce: state.nonce,
+        networkDescriptor: state.networkDescriptor,
+        tokenName: state.name,
+        tokenSymbol: state.symbol,
+        tokenDecimals: state.decimals,
+        validators: [userOne, userTwo, userFour],
+      });
+
+      const expectedAddress = ethers.utils.getContractAddress({ from: state.bridgeBank.address, nonce: 1 });
+      await expect(state.cosmosBridge
+        .connect(userOne)
+        .submitProphecyClaimAggregatedSigs(
+            digest,
+            claimData,
+            signatures
+        )).to.emit(state.cosmosBridge, 'LogNewBridgeTokenCreated')
+        .withArgs(18, state.networkDescriptor, state.name, state.symbol, state.token.address, expectedAddress);
+
+      const newlyCreatedTokenAddress = await state.cosmosBridge.sourceAddressToDestinationAddress(state.token.address);
+      expect(newlyCreatedTokenAddress).to.be.equal(expectedAddress);
+
+      // Everything again, but this time submitProphecyClaimAggregatedSigs should NOT emit the event
+      const { digest: digest2, claimData: claimData2, signatures: signatures2 } = await getValidClaim({
+        state,
+        sender: state.sender,
+        senderSequence: state.senderSequence + 1,
+        recipientAddress: state.recipient,
+        tokenAddress: state.token.address,
+        amount: state.amount,
+        isDoublePeg: true,
+        nonce: state.nonce + 1,
+        networkDescriptor: state.networkDescriptor,
+        tokenName: state.name,
+        tokenSymbol: state.symbol,
+        tokenDecimals: state.decimals,
+        validators: [userOne, userTwo, userFour],
+      });
+
+      await expect(state.cosmosBridge
```

Comment:
Earlier in this test, you say that this time it shouldn't emit an event, but here, you are testing that an event was emitted. I'm a bit confused.